server/scripts/examples-ml5.js

```diff
@@ -0,0 +1,457 @@
+
+  output = output.map(async item => fetchSketchContent(item));
+
+  output = await Q.all(output);
+
+  return output;
+}
+
+/**
+ * STEP 6
+ * Remove existing projects, then fill the db
+ * @param {*} filledProjectList
+ * @param {*} user
+ */
+async function createProjectsInP5User(filledProjectList, user) {
+  const userProjects = await Project.find({ user: user._id });
+  const removeProjects = userProjects.map(async project => Project.remove({ _id: project._id }));
+  await Q.all(removeProjects);
+  console.log('deleted old projects!');
+
+  const newProjects = filledProjectList.map(async (project) => {
+    const item = new Project(project);
+    console.log(`saving ${project.name}`);
+    await item.save();
+  });
+  await Q.all(newProjects);
+  console.log(`Projects saved to User: ${editorUsername}!`);
+}
+
+/**
+ * STEP 0
+ * CHECK if user exists, ifnot create one
+ *
+ */
+async function checkP5User() {
+  const user = await User.findOne({ username: editorUsername });
+
+  if (!user) {
+    const ml5user = new User({
+      username: editorUsername,
+      email: process.env.ML5_EXAMPLES_EMAIL,
+      password: process.env.ML5_EXAMPLES_PASS
+    });
+
+    await ml5user.save((saveErr) => {
+      if (saveErr) throw saveErr;
+      console.log(`Created a user p5${ml5user}`);
+    });
+  }
+}
+
+
+/**
+ * ---------------------------------------------------------
+ * --------------------- main ------------------------------
+ * ---------------------------------------------------------
+ */
+
+/**
+ * MAKE
+ * Get all the sketches from the ml5-examples repo
+ * Get the p5 examples
+ * Dive down into each sketch and get all the files
+ * Format the sketch files to be save to the db
+ * Delete existing and save
+ */
+async function make() {
+  await checkP5User();
+  // Get the user
+  const user = await User.findOne({
+    username: editorUsername
+  });
+  // Get the categories and their examples
+  const categories = await getCategories();
+  const categoryExamples = await getCategoryExamples(categories);
+  const examplesWithResourceTree = await traverseSketchTreeAll(categoryExamples);
+  const formattedSketchList = formatSketchForStorageAll(examplesWithResourceTree, user);
+  const filledProjectList = await fetchSketchContentAll(formattedSketchList);
+  await createProjectsInP5User(filledProjectList, user);
+  console.log('done!');
+  process.exit();
+}
+
+/**
+ * TEST - same as make except reads from file for testing purposes
+ * Get all the sketches from the ml5-examples repo
+ * Get the p5 examples
+ * Dive down into each sketch and get all the files
+ * Format the sketch files to be save to the db
+ * Delete existing and save
+ */
+async function test() {
+  await checkP5User();
+  // Get the user
+  const user = await User.findOne({
+    username: editorUsername
+  });
+
+  // read from file while testing
+  const examplesWithResourceTree = JSON.parse(fs.readFileSync('./ml5-examplesWithResourceTree.json'));
```

Comment:
@joeyklee Do you remember what the structure of this file was and/or how did you generate the JSON? I initially tried copying over the [`examples.json`](https://github.com/ml5js/ml5-library/blob/main/examples/examples.json) file we use in ml5.js, but it looks like this is probably a list that is formatted in a structure more closely related to how the web editor represents sketches.

Reply:
@bomanimc - Ah yes, these were files I'd saved locally while testing locally. These files would mirror structure of the github API results while traversing the folder directories. I will follow up via email w/ the link to the data. 

(The `examples.json` file unfortunately doesn't solve this issue for us since the p5 web editor script would be pulling all of the files from Github.)